src/Symfony/Component/HttpKernel/DependencyInjection/ControllerArgumentValueResolverPass.php

```diff
@@ -47,6 +50,11 @@ public function process(ContainerBuilder $container)
 
         $resolvers = $this->findAndSortTaggedServices($this->argumentValueResolverTag, $container);
 
+        if ($container->getParameter('kernel.debug')) {
+            $container->register($this->notTaggedControllerResolverServiceId, NotTaggedControllerValueResolver::class);
+            $resolvers[] = new Reference($this->notTaggedControllerResolverServiceId);
+        }
```

Comment:
I would not register it this way. Instead, what about moving the definition in the `src/Symfony/Bundle/FrameworkBundle/Resources/config/debug.xml` file instead?

Reply:
Registration of resolver moved to `src/Symfony/Bundle/FrameworkBundle/Resources/config/debug.xml`.